command/debug.go

```diff
@@ -0,0 +1,945 @@
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to create client to connect to Vault: %s", err)
+	}
+	if _, err := client.Sys().Health(); err != nil {
+		return nil, "", fmt.Errorf("unable to connect to the server: %s", err)
+	}
+
+	captureTime := time.Now().UTC()
+	if len(c.flagOutput) == 0 {
+		formattedTime := captureTime.Format(fileFriendlyTimeFormat)
+		c.flagOutput = fmt.Sprintf("vault-debug-%s", formattedTime)
+	}
+
+	// Strip trailing slash before proceeding
+	c.flagOutput = strings.TrimSuffix(c.flagOutput, "/")
+
+	// If compression is enabled, trim the extension so that the files are
+	// written to a directory even if compression somehow fails. We ensure the
+	// extension during compression. We also prevent overwriting if the file
+	// already exists.
+	dstOutputFile := c.flagOutput
+	if c.flagCompress {
+		if !strings.HasSuffix(dstOutputFile, ".tar.gz") && !strings.HasSuffix(dstOutputFile, ".tgz") {
+			dstOutputFile = dstOutputFile + debugCompressionExt
+		}
+
+		// Ensure that the file doesn't already exist, and ensure that we always
+		// trim the extension from flagOutput since we'll be progressively
+		// writing to that.
+		if _, err := os.Stat(dstOutputFile); os.IsNotExist(err) {
+			c.flagOutput = strings.TrimSuffix(c.flagOutput, ".tar.gz")
+			c.flagOutput = strings.TrimSuffix(c.flagOutput, ".tgz")
+		} else {
+			return nil, "", fmt.Errorf("output file already exists: %s", dstOutputFile)
+		}
+	}
+
+	// Stat check the directory to ensure we don't override any existing data.
+	if _, err := os.Stat(c.flagOutput); os.IsNotExist(err) {
+		err := os.MkdirAll(c.flagOutput, 0755)
+		if err != nil {
+			return nil, "", fmt.Errorf("unable to create output directory: %s", err)
+		}
+	} else {
+		return nil, "", fmt.Errorf("output directory already exists: %s", c.flagOutput)
+	}
+
+	// Instantiate the channels for polling
+	c.metricsCh = make(chan map[string]interface{})
+	c.serverStatusCh = make(chan map[string]interface{})
+	c.replicationStatusCh = make(chan map[string]interface{})
+	c.hostInfoCh = make(chan map[string]interface{})
+
+	// Populate initial index fields
+	c.debugIndex = &debugIndex{
+		VaultAddress:    client.Address(),
+		ClientVersion:   version.GetVersion().VersionNumber(),
+		Compress:        c.flagCompress,
+		Duration:        int(c.flagDuration.Seconds()),
+		Interval:        int(c.flagInterval.Seconds()),
+		MetricsInterval: int(c.flagMetricsInterval.Seconds()),
+		RawArgs:         rawArgs,
+		Version:         debugIndexVersion,
+		Targets:         c.flagTargets,
+		Timestamp:       captureTime,
+		Errors:          []*captureError{},
+	}
+
+	return client, dstOutputFile, nil
+}
+
+func (c *DebugCommand) defaultTargets() []string {
+	return []string{"config", "host", "metrics", "pprof", "replication-status", "server-status"}
+}
+
+func (c *DebugCommand) captureStaticTargets(client *api.Client) error {
+	// Capture configuration state
+	if strutil.StrListContains(c.flagTargets, "config") {
```

Comment:
I'd be inclined to move the `if` outside of the method.  Failing that, can you invert it and do an early return so that we have less indentation?

Reply:
This logic would not be applicable if other static targets were to be added, so in the spirit of keeping it to the same pattern as intervalCapture, I've done it this way.